web/app.js

```diff
@@ -329,16 +332,18 @@ var PDFViewerApplication = {
 
       this.pdfViewer.setFindController(this.findController);
 
-      // FIXME better PDFFindBar constructor parameters
+      // TODO: improve `PDFFindBar` constructor parameter passing
       let findBarConfig = Object.create(appConfig.findBar);
       findBarConfig.findController = this.findController;
       findBarConfig.eventBus = eventBus;
       this.findBar = new PDFFindBar(findBarConfig);
 
-      this.overlayManager = OverlayManager;
-
+      // TODO: improve `PDFDocumentProperties` constructor parameter passing
+      let documentPropertiesConfig =
+        Object.create(appConfig.documentProperties);
+      documentPropertiesConfig.overlayManager = this.overlayManager;
       this.pdfDocumentProperties =
-        new PDFDocumentProperties(appConfig.documentProperties);
+        new PDFDocumentProperties(documentPropertiesConfig);
```

Comment:
What you could perhaps do instead (here and elsewhere in the patch), to avoid adding even more of these hacks, is something similar to what PR #8394 does; see https://github.com/mozilla/pdf.js/pull/8394/files#diff-529d1853ee1bba753a0fcb40ea778723R372.